17.md

```diff
@@ -0,0 +1,154 @@
+NIP-17
+======
+
+Private Direct Messages
+-----------------------
+
+`draft` `optional`
+
+This NIP defines an encrypted direct messaging scheme using [NIP-44](44.md) encryption and [NIP-59](59.md) seals and gift wraps. 
+
+## Direct Message Kind
+
+Kind `14` is a chat message. `p` tags identify one or more receivers of the message.
+
+```js
+{
+  "id": "<usual hash>",
+  "pubkey": "<sender-pubkey>",
+  "created_at": now(),
+  "kind": 14,
+  "tags": [ 
+    ["p", "<receiver-1-pubkey>", "<relay-url>"],
+    ["p", "<receiver-2-pubkey>", "<relay-url>"],
+    ["e", "<kind-14-id>", "<relay-url>", "reply"] // if this is a reply
+    ["subject", "<conversation-title>"],
+    ...
+  ],
+  "content": "<message-in-plain-text>",
+}
+```
+
+`.content` MUST be plain text. Fields `id` and `created_at` are required. 
+
+Tags that mention, quote and assemble threading structures MUST follow [NIP-10](10.md). 
+
+Kind `14`s MUST never be signed. If it is signed, the message might leak to relays and become **fully public**.
+
+## Chat Rooms
+
+The set of `pubkey` + `p` tags defines a chat room. If a new `p` tag is added or a current one is removed, a new room is created with clean message history. 
+
+Clients SHOULD render messages of the same room in a continuous thread. 
+
+An optional `subject` tag defines the current name/topic of the conversation. Any member can change the topic by simply submitting a new `subject` to an existing `pubkey` + `p`-tags room. There is no need to send `subject` in every message. The newest `subject` in the thread is the subject of the conversation. 
+
+## Encrypting
+
+Following [NIP-59](59.md), the **unsigned** `kind:14` chat message must be sealed (`kind:13`) and then gift-wrapped (`kind:1059`) to each receiver and the sender individually.
+
+```js
+{
+  "id": "<usual hash>",
+  "pubkey": randomPublicKey,
+  "created_at": randomTimeUpTo2DaysInThePast(), 
+  "kind": 1059, // gift wrap
+  "tags": [
+    ["p", receiverPublicKey, "<relay-url>"] // receiver
+  ],
+  "content": nip44Encrypt(
+    {
+      "id": "<usual hash>",
+      "pubkey": senderPublicKey,
+      "created_at": randomTimeUpTo2DaysInThePast(), 
+      "kind": 13, // seal
+      "tags": [], // no tags
+      "content": nip44Encrypt(unsignedKind14, senderPrivateKey, receiverPublicKey),
+      "sig": "<signed by senderPrivateKey>"
+    },
+    randomPrivateKey, receiverPublicKey
+  ),
+  "sig": "<signed by randomPrivateKey>"
+}
+```
+
+The encryption algorithm MUST use the latest version of [NIP-44](44.md). 
+
+Clients MUST verify if pubkey of the `kind:13` is the same pubkey on the `kind:14`, otherwise any sender can impersonate others by simply changing the pubkey on `kind:14`.
+
+Clients SHOULD randomize `created_at` in up to two days in the past in both the seal and the gift wrap to make sure grouping by `created_at` doesn't reveal any metadata. 
```

Comment:
Won't some clients miss the gift wrapped event if `created_at` is up to two days in the past? Some clients may use `created_at` set to `now` as a cursor to avoid fetching old events. Or do clients just need know that they need to fetch up to two days in the past for the gift wrap kind 1059?

Reply:
Yep, always fetch 2 days prior to the last fetch. Hopefully we get some Sync capabilities in the future.